pkg/server/server.go

```diff
@@ -0,0 +1,629 @@
+	//then we have to call the below line. Determine if we need to check this
+	//or if we are assuming all pod template specs have prior annotations.
+	// tempPTS.Annotations = make(map[string]string)
+	tempPTS.Annotations["trafficHosts"] = tempJSON.TrafficHosts
+
+	template := extensions.Deployment{
+		ObjectMeta: api.ObjectMeta{
+			Name: tempJSON.DeploymentName,
+		},
+		Spec: extensions.DeploymentSpec{
+			Replicas: tempJSON.Replicas,
+			Selector: &unversioned.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": tempPTS.Labels["app"],
+				},
+			},
+			Template: *tempPTS,
+		},
+	}
+
+	//Create Deployment
+	dep, err := client.Deployments(pathVars["environmentGroupID"] + "-" + pathVars["environment"]).Create(&template)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error creating deployment: %v\n", err)
+		return
+	}
+	js, err := json.Marshal(dep)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error marshalling deployment: %v\n", err)
+	}
+
+	w.WriteHeader(201)
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(js)
+
+	fmt.Printf("Created Deployment: %v\n", dep.GetName())
+}
+
+//getDeployment returns a deployment matching the given environmentGroupID, environmentName, and deploymentName
+func getDeployment(w http.ResponseWriter, r *http.Request) {
+	pathVars := mux.Vars(r)
+
+	depList, err := client.Deployments(pathVars["environmentGroupID"] + "-" + pathVars["environment"]).List(api.ListOptions{
+		LabelSelector: labels.Everything(),
+	})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error retrieving deployment list: %v\n", err)
+		return
+	}
+	for _, value := range depList.Items {
+		if value.GetName() == pathVars["deployment"] {
+			js, err := json.Marshal(value)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				fmt.Printf("Error marshalling deployment: %v\n", err)
+			}
+
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "application/json")
+			w.Write(js)
+			fmt.Printf("Got Deployment: %v\n", value.GetName())
+		}
+	}
+
+}
+
+//TODO: Allow modifying the PTS
+//updateDeployment updates a deployment matching the given environmentGroupID, environmentName, and deploymentName
+func updateDeployment(w http.ResponseWriter, r *http.Request) {
+	pathVars := mux.Vars(r)
+
+	//Struct to put JSON into
+	type deploymentPatch struct {
+		TrafficHosts string `json:"trafficHosts"`
+		Replicas     int    `json:"Replicas"`
+		PtsURL       string `json:"ptsURL"`
+	}
+	//Decode passed JSON body
+	decoder := json.NewDecoder(r.Body)
+	var tempJSON deploymentPatch
+	err := decoder.Decode(&tempJSON)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error decoding JSON Body: %v\n", err)
+		return
+	}
+
+	//Get JSON from url
+	httpClient := &http.Client{}
```

Comment:
Gathering the Pod template spec seems to be code that is reused so it could be in a helper to avoid duplicate code.